vendor/bat-native-ads/src/bat/ads/internal/eligible_ads/ad_notifications/eligible_ad_notifications_unittest.cc

```diff
@@ -61,11 +58,31 @@ class BatAdsEligibleAdNotificationsTest : public UnitTestBase {
   }
 
   void Save(const CreativeAdNotificationList& creative_ad_notifications) {
-    database_table_->Save(creative_ad_notifications,
-                          [](const bool success) { ASSERT_TRUE(success); });
+    creative_ad_notifications_table_->Save(
+        creative_ad_notifications,
+        [](const bool success) { ASSERT_TRUE(success); });
+  }
+
+  void Log(const CreativeAdNotificationInfo& creative_ad_notification,
+           int hours_ago) {
+    AdEventInfo ad_event;
+    ad_event.uuid = base::GenerateGUID();
+    ad_event.type = AdType::kAdNotification;
+    ad_event.confirmation_type = ConfirmationType::kViewed;
+    ad_event.campaign_id = creative_ad_notification.campaign_id;
+    ad_event.creative_set_id = creative_ad_notification.creative_set_id;
+    ad_event.creative_instance_id =
+        creative_ad_notification.creative_instance_id;
+    ad_event.advertiser_id = creative_ad_notification.advertiser_id;
+    base::Time timestamp =
+        base::Time::Now() - base::TimeDelta::FromHours(hours_ago);
+    ad_event.timestamp = static_cast<int64_t>(timestamp.ToDoubleT());
+
+    LogAdEvent(ad_event, [](const bool success) { ASSERT_TRUE(success); });
```

Comment:
Decouple to a separate `unittest_util` (see other comments). We could also pass the `type`, `confirmation_type` and `base::TimeDelta` instead of int